syntax = "proto3";

package messenger;

option java_package = "com.azaat.grpcchatclient";
option java_outer_classname = "MessengerProto";
  
message MessengerMessage {
    string message = 1;
}

message Empty {}

message MessengerNameRequest {
    string name = 1;
}

message MessengerNameResponse {
    string name = 1;
    bool connected = 2;
}

service Messenger {
    rpc getMessage (MessengerMessage) returns (Empty);
    rpc sendMessage (Empty) returns (stream MessengerMessage);
    rpc startMessaging (MessengerNameRequest) returns (MessengerNameResponse);
    rpc stopMessaging (Empty) returns (Empty);
}